src/main/java/ch/wisv/areafiftylan/seats/service/SeatServiceImpl.java

```diff
@@ -74,43 +76,49 @@ public SeatmapResponse getSeatGroupByName(String groupname) {
     }
 
     @Override
-    public boolean reserveSeatForTicket(String groupname, int seatnumber, Long ticketId) {
-        Ticket ticket = ticketRepository.findOne(ticketId);
-
-        if (!ticket.isValid()) {
-            throw new InvalidTicketException("Unable to reserve seat for an invalid Ticket");
-        }
-
-        Optional<Seat> previousSeat = seatRepository.findByTicketId(ticketId);
-
-        // We can only reserve one seat at a time, to prevent the extremely unlikely event of simultaneous requests
-        synchronized (seatReservationLock) {
-            Seat seat = seatRepository.findBySeatGroupAndSeatNumber(groupname, seatnumber);
-            previousSeat.ifPresent(s -> s.setTicket(null));
-            if (!seat.isTaken()) {
-                seat.setTicket(ticket);
-                seatRepository.saveAndFlush(seat);
-                return true;
-            } else {
-                return false;
-            }
+    public boolean reserveSeatForTicket(String seatGroup, int seatNumber, Long ticketId) {
+        Seat seat = seatRepository.findBySeatGroupAndSeatNumber(seatGroup, seatNumber);
```

Comment:
This already exists: `getSeatBySeatGroupAndSeatNumber()`, only without exception throwing